go/cmd/dolt/commands/log_graph.go

```diff
@@ -0,0 +1,536 @@
+*			a. the vertical distance is greater than the horizontal distance	    b. the horizontal distance is greater than the vertical distance
+*              * child commit 					 										* --- child commit
+* 				 \								     										  \
+* 				  \                                 									       \
+* 				  |                                     									    \
+* 				  * parent commit                          										 * parent commit
+*
+* ------------
+* Sample output
+* -------------
+*
+* The following is an example of the output.
+* ------------
+
+*   commit skcm452jteobhaqb2pngc8rbc37p0p0d(HEAD -> main)
+|\  Merge: qcfirrj62mlvv6002bq6m9ep16t4ej61 tnenj0ntvhue15v3veq98ng4pi2vvqta
+| | Author: liuliu <liu@dolthub.com>
+| | Date: Thu Aug 01 10:23:22 -0700 2024
+| |
+| |     Merge branch 'd' into main
+| |
+* |   commit qcfirrj62mlvv6002bq6m9ep16t4ej61
+|\|   Merge: nlmgt75rl0a4j6ff8qp8afj2t70l3dfo 78vpv0ij6mr788vnm3k8r36m4k519ob1
+| \   Author: liuliu <liu@dolthub.com>
+| |\  Date: Thu Aug 01 10:23:19 -0700 2024
+| | |
+| | |   Merge branch 'c' into main
+| | |
+* | |   commit nlmgt75rl0a4j6ff8qp8afj2t70l3dfo
+|\| |   Merge: l49as4tsfnph2v5edopu8a5cljs8vbc6 oj2frj1292visbj1ve14amhpcq0hn959
+| \ |   Author: liuliu <liu@dolthub.com>
+| |\|   Date: Thu Aug 01 10:23:16 -0700 2024
+| | \
+| | |\          Merge branch 'b' into main
+| | | |
+* | | | commit l49as4tsfnph2v5edopu8a5cljs8vbc6
+|\| | | Merge: dk7dd4v2fhj40q3bbv0no9s1vpq41qef 7723gb3u9b125gum67dgier1t67s5md0
+| \ | | Author: liuliu <liu@dolthub.com>
+| |\| | Date: Thu Aug 01 10:22:59 -0700 2024
+| | \ |
+| | |\|         Merge branch 'a' into main
+| | | \
+* | | |\  commit dk7dd4v2fhj40q3bbv0no9s1vpq41qef
+| | | | | Author: liuliu <liu@dolthub.com>
+| | | | | Date: Thu Aug 01 10:22:54 -0700 2024
+| | | | |
+| | | | |       change
+| | | | |
+| * | | | commit tnenj0ntvhue15v3veq98ng4pi2vvqta(d)
+| | | | | Author: liuliu <liu@dolthub.com>
+| | | | | Date: Thu Aug 01 10:22:43 -0700 2024
+| | | | |
+| | | | |       d
+| | | | |
+| | * | | commit 78vpv0ij6mr788vnm3k8r36m4k519ob1(c)
+| | | | | Author: liuliu <liu@dolthub.com>
+| | | | | Date: Thu Aug 01 10:22:30 -0700 2024
+| | | | |
+| | | | |       c
+| | | | |
+| | | * | commit oj2frj1292visbj1ve14amhpcq0hn959(b)
+| | | | | Author: liuliu <liu@dolthub.com>
+| | | | | Date: Thu Aug 01 10:22:14 -0700 2024
+| | | | |
+| | | | |       b
+| | | | |
+| | | | * commit 7723gb3u9b125gum67dgier1t67s5md0(a)
+| | |/ /  Author: liuliu <liu@dolthub.com>
+| | / /   Date: Thu Aug 01 10:21:57 -0700 2024
+| |/ /
+| / /           a
+|/ /
+*-- commit lqoem7sk4l1qjbdlf8k973o89bki0vlk
+|   Author: liuliu <liu@dolthub.com>
+|   Date: Thu Aug 01 10:21:37 -0700 2024
+|
+|       main
+|
+* commit 87bg8tfrvjo8cfbak92flu595p5e4bbl
+  Author: liuliu <liu@dolthub.com>
+  Date: Thu Aug 01 10:20:44 -0700 2024
+
+       Initialize data repository
+
+* ------------
+*
+*/
+
+type CommitInfoWithChildren struct {
```

Comment:
This type is only used in the package, so make it start with a lower case letter (makes it package private)